### PR TITLE
Pagination documentation update

### PR DESCRIPTION
**Pull request by joshfix**
**Related Issue(s):** https://github.com/radiantearth/stac-spec/issues/716

**Proposed Changes:**

1. Documentation updates to clarify how to implement new pagination mechanism for POST requests

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: api-spec/api-spec.md [hunk 1]
```diff
@@ -104,3 +156,57 @@ All Extensions **should** use attribute names qualified from the root of Item, r
...
+execute a subsequent request for the next page of results.
+
+The following fields have been added to the `link` object specification for the API spec:
+
+| Parameter | Type             | Description |
+| --------- | ---------------- | ----------- |
+| method    | string           | One of `GET` or `POST` to indicate the HTTP method of the request |
+| headers   | object           | A dictionary of header values that should be included in the next request |
+| body      | object           | A JSON object containing fields/values that should be included in the body of the next request |
+| merge     | boolean          | If true, the headers/body fields in the next link should be merged into the original request for the next request |
```
Comment (m-mohr) [inline api-spec/api-spec.md]: We should mention the default value for merge.

File: api-spec/api-spec.md [hunk 2]
```diff
@@ -104,3 +156,57 @@ All Extensions **should** use attribute names qualified from the root of Item, r
...
+OAFeat does not support POST requests for searches, however the STAC API spec does. Hypermedia links are not designed 
+for anything other than GET requests, so providing a next link for a POST search request becomes problematic. STAC has 
+decided to extend the `link` object to support additional fields that provide hints to the client as to how it should 
+execute a subsequent request for the next page of results.
+
+The following fields have been added to the `link` object specification for the API spec:
+
+| Parameter | Type             | Description |
+| --------- | ---------------- | ----------- |
+| method    | string           | One of `GET` or `POST` to indicate the HTTP method of the request |
```
Comment (m-mohr) [inline api-spec/api-spec.md]:
```suggestion
| method    | string           | The HTTP method of the request, usually `GET` or `POST`. Defaults to `GET` |
```

File: api-spec/api-spec.md (hunk 1 above)
Comment (m-mohr) [inline api-spec/api-spec.md]:
```suggestion
| merge     | boolean          | If `true`, the headers/body fields in the `next` link should be merged into the original request and be sent combined in the next request. Defaults to `false` | 
```

File: api-spec/api-spec.md [hunk 3]
```diff
@@ -104,3 +156,57 @@ All Extensions **should** use attribute names qualified from the root of Item, r
...
+
+### Paging Extension
+
+OGC API supports paging through hypermedia links and STAC follows the same pattern for the cross collection search. For 
+GET requests, a link with `rel` type `next` is supplied.  This link may contain any URL parameter that is necessary 
+for the implementation to understand how to provide the next page of results, eg: `page`, `next`, `token`, etc.  The 
+parameter name is defined by the implementor and is not necessarily part of the API specification.  For example:
+
+```json
+{
```
Comment (m-mohr) [inline api-spec/api-spec.md]: I guess it would be good to also have a more complex example with method, headers etc, too.

Comment (joshfix) [reply api-spec/api-spec.md]: I cleaned up the examples and added an additional example using headers.

**Pull request merged**